exporting/read_config.c

```diff
@@ -220,15 +210,54 @@ struct engine *read_exporting_config()
 
     freez(filename);
 
+#define prometheus_config_get(name, value)                                                                             \
+    appconfig_get(                                                                                                     \
+        &exporting_config, CONFIG_SECTION_PROMETHEUS, name,                                                            \
+        appconfig_get(&exporting_config, CONFIG_SECTION_EXPORTING, name, value))
+#define prometheus_config_get_number(name, value)                                                                      \
+    appconfig_get_number(                                                                                              \
+        &exporting_config, CONFIG_SECTION_PROMETHEUS, name,                                                            \
+        appconfig_get_number(&exporting_config, CONFIG_SECTION_EXPORTING, name, value))
+#define prometheus_config_get_boolean(name, value)                                                                     \
+    appconfig_get_boolean(                                                                                             \
+        &exporting_config, CONFIG_SECTION_PROMETHEUS, name,                                                            \
+        appconfig_get_boolean(&exporting_config, CONFIG_SECTION_EXPORTING, name, value))
+
+    if (!prometheus_exporter_instance) {
+        prometheus_exporter_instance = callocz(1, sizeof(struct instance));
+
+        prometheus_exporter_instance->config.update_every =
+            prometheus_config_get_number(EXPORTING_UPDATE_EVERY_OPTION_NAME, EXPORTING_UPDATE_EVERY_DEFAULT);
+
+        if (prometheus_config_get_boolean("send names instead of ids", CONFIG_BOOLEAN_YES))
+            prometheus_exporter_instance->config.options |= EXPORTING_OPTION_SEND_NAMES;
+        else
+            prometheus_exporter_instance->config.options &= ~EXPORTING_OPTION_SEND_NAMES;
+
+        if (prometheus_config_get_boolean("send configured labels", CONFIG_BOOLEAN_YES))
+            prometheus_exporter_instance->config.options |= EXPORTING_OPTION_SEND_CONFIGURED_LABELS;
+        else
+            prometheus_exporter_instance->config.options &= ~EXPORTING_OPTION_SEND_CONFIGURED_LABELS;
+
+        if (prometheus_config_get_boolean("send automatic labels", CONFIG_BOOLEAN_NO))
```

Comment:
Only to confirm, this is the unique condition that by default won't set by default the flag, for all others you are always enabling, is this expected? Because you are sending `configured labels`, but you are ignoring the `automatic labels`.

Reply:
No. Every connector has [disabled automatic labels](https://github.com/netdata/netdata/blob/master/exporting/read_config.c#L336) by default.